soc/silabs/silabs_siwx917/siwg917/nwp_init.c

```diff
@@ -0,0 +1,118 @@
+	 SL_SI91X_BLE_MAX_NBR_ATT_REC(RSI_BLE_MAX_NBR_ATT_REC) |                                   \
+	 SL_SI91X_BLE_PWR_INX(RSI_BLE_PWR_INX) |                                                   \
+	 SL_SI91X_BLE_PWR_SAVE_OPTIONS(RSI_BLE_PWR_SAVE_OPTIONS) |                                 \
+	 SL_SI91X_916_BLE_COMPATIBLE_FEAT_ENABLE | SL_SI91X_FEAT_BLE_CUSTOM_FEAT_EXTENTION_VALID)
+
+#define NWP_INIT_BLE_EXT_FEATURE_BIT_MAP_BLE                                                       \
+	(SL_SI91X_BLE_NUM_CONN_EVENTS(RSI_BLE_NUM_CONN_EVENTS) |                                   \
+	 SL_SI91X_BLE_NUM_REC_BYTES(RSI_BLE_NUM_REC_BYTES) | SL_SI91X_BLE_ENABLE_ADV_EXTN |        \
+	 SL_SI91X_BLE_AE_MAX_ADV_SETS(RSI_BLE_AE_MAX_ADV_SETS))
+
+#define NWP_INIT_EXT_CUSTOM_FEATURE_BIT_MAP_COMMON                                                 \
+	(MEMORY_CONFIG | SL_SI91X_EXT_FEAT_XTAL_CLK |                                              \
+	 SL_SI91X_EXT_FEAT_FRONT_END_SWITCH_PINS_ULP_GPIO_4_5_0)
+
+#if CONFIG_WIFI_SIWX917 && CONFIG_BT_SIWX917
+#define NWP_INIT_COEX_MODE SL_SI91X_WLAN_BLE_MODE
+#define NWP_INIT_TCP_IP_FEATURE_BIT_MAP                                                            \
+	(NWP_INIT_TCP_IP_FEATURE_BIT_MAP_WIFI | NWP_INIT_TCP_IP_FEATURE_BIT_MAP_BLE)
+#define NWP_INIT_EXT_CUSTOM_FEATURE_BIT_MAP                                                        \
+	(NWP_INIT_EXT_CUSTOM_FEATURE_BIT_MAP_COMMON | SL_SI91X_EXT_FEAT_BT_CUSTOM_FEAT_ENABLE)
+#define NWP_INIT_BT_FEATURE_BIT_MAP         NWP_INIT_BT_FEATURE_BIT_MAP_BLE
+#define NWP_INIT_EXT_TCP_IP_FEATURE_BIT_MAP NWP_INIT_EXT_TCP_IP_FEATURE_BIT_MAP_WIFI
+#define NWP_INIT_BLE_FEATURE_BIT_MAP        NWP_INIT_BLE_FEATURE_BIT_MAP_BLE
+#define NWP_INIT_BLE_EXT_FEATURE_BIT_MAP    NWP_INIT_BLE_EXT_FEATURE_BIT_MAP_BLE
+#elif CONFIG_WIFI_SIWX917
+#define NWP_INIT_COEX_MODE                  SL_SI91X_WLAN_ONLY_MODE
+#define NWP_INIT_TCP_IP_FEATURE_BIT_MAP     NWP_INIT_TCP_IP_FEATURE_BIT_MAP_WIFI
+#define NWP_INIT_EXT_CUSTOM_FEATURE_BIT_MAP NWP_INIT_EXT_CUSTOM_FEATURE_BIT_MAP_COMMON
+#define NWP_INIT_BT_FEATURE_BIT_MAP         0
+#define NWP_INIT_EXT_TCP_IP_FEATURE_BIT_MAP NWP_INIT_EXT_TCP_IP_FEATURE_BIT_MAP_WIFI
+#define NWP_INIT_BLE_FEATURE_BIT_MAP        0
+#define NWP_INIT_BLE_EXT_FEATURE_BIT_MAP    0
+#elif CONFIG_BT_SIWX917
+#define NWP_INIT_COEX_MODE              SL_SI91X_BLE_MODE
+#define NWP_INIT_TCP_IP_FEATURE_BIT_MAP NWP_INIT_TCP_IP_FEATURE_BIT_MAP_BLE
+#define NWP_INIT_EXT_CUSTOM_FEATURE_BIT_MAP                                                        \
+	(NWP_INIT_EXT_CUSTOM_FEATURE_BIT_MAP_COMMON | SL_SI91X_EXT_FEAT_BT_CUSTOM_FEAT_ENABLE)
+#define NWP_INIT_BT_FEATURE_BIT_MAP         NWP_INIT_BT_FEATURE_BIT_MAP_BLE
+#define NWP_INIT_EXT_TCP_IP_FEATURE_BIT_MAP SL_SI91X_CONFIG_FEAT_EXTENTION_VALID
+#define NWP_INIT_BLE_FEATURE_BIT_MAP        NWP_INIT_BLE_FEATURE_BIT_MAP_BLE
+#define NWP_INIT_BLE_EXT_FEATURE_BIT_MAP    NWP_INIT_BLE_EXT_FEATURE_BIT_MAP_BLE
+#else
+#error "No configuration selected"
+#endif
```

Comment:
I feel these definitions are difficult to read. I think it would be better to fill `network_config` during runtime.

Reply:
I agree it's not easy to read but I'm preferring to solve something compile time over run time.
On the other hand in case of run time configuration the silabs_siwx917_nwp_init() function will be harder to read.
However if you insist on this, I'll change it.

Reply:
If you do it at runtime, I think you will be able to remove two layers of indirection (all the `NWP_INIT_*` definition). In addition, I believe the semantic of the options will be appear more clearly.

Reply:
Don't forget to use `IS_ENABLED()` to keep code readable.